src/box/xrow_update_array.c

```diff
@@ -370,8 +370,18 @@ xrow_update_op_do_array_insert(struct xrow_update_op *op,
 
 	struct xrow_update_rope *rope = field->array.rope;
 	uint32_t size = xrow_update_rope_size(rope);
-	assert(size <= BOX_FIELD_MAX);
-	if (size == BOX_FIELD_MAX) {
+	/*
+	 * TODO: if the error injection code hits performance, we must put it
+	 * under conditional compilation.
+	 */
```

Comment:
AFAIU for release builds the check is compiled out, because `errinj()` returns NULL. Since we only care about performance of release builds, I think you should remove this comment.

Reply:
Agreed on the assumption that compilers eliminate such dead code.

Removed this comment.

Reply:
Proof: https://godbolt.org/z/bPs6EY9ec